x/cert/client/cli/tx.go

```diff
@@ -141,31 +141,20 @@ func GetCmdIssueCertificate() *cobra.Command {
 				return err
 			}
 
+			compiler, bytecodeHash := "", ""
 			certificateTypeString := strings.ToLower(args[0])
-			switch certificateTypeString {
-			case "compilation":
-				contentType := types.RequestContentTypeFromString(args[1])
-				if contentType != types.RequestContentTypeSourceCodeHash {
-					return types.ErrInvalidRequestContentType
-				}
-				compiler, bytecodeHash, description, err := parseCertifyCompilationFlags()
+			if certificateTypeString == "compilation" {
+				compiler, bytecodeHash, err = parseCertifyCompilationFlags()
 				if err != nil {
 					return err
 				}
-				msg := types.NewMsgCertifyCompilation(args[2], compiler, bytecodeHash, description, from)
-				if err := msg.ValidateBasic(); err != nil {
-					return err
-				}
-				return tx.GenerateOrBroadcastTxWithFactory(cliCtx, txf, msg)
-
-			default:
-				description := viper.GetString(FlagDescription)
-				msg := types.NewMsgCertifyGeneral(certificateTypeString, args[1], args[2], description, from)
-				if err := msg.ValidateBasic(); err != nil {
-					return err
-				}
-				return tx.GenerateOrBroadcastTxWithFactory(cliCtx, txf, msg)
 			}
+			description := viper.GetString(FlagDescription)
+			msg := types.NewMsgIssueCertificate(args[0], args[1], compiler, bytecodeHash, description, from)
```

Comment:
I suppose we can just `AssembleContent` here and have `MsgIssueCertificate` just contain `Content` interface type. That's how proposal submission txs work. What do you think?

Reply:
I think it's a good idea to assemble content here, it'll keep the msg type simple.